src/docbuilder/metadata/.gitignore

```diff
@@ -0,0 +1 @@
+Cargo.lock
```

Comment:
We should make the metadata crate part of a workspace instead of ignoring the `Cargo.lock`.

Reply:
This is also preventing running the metadata tests with `cargo test --workspace`. We're not executing them at all on CI at the moment.

Reply:
Can do, but I found a bug in clippy while changing this - it will ignore `--all` and only run clippy on the top-level directory. I'll change CI to run it once for each crate.

Reply:
Err this was just me misunderstanding how clippy works, it does listen to `--all`.